### PR TITLE
Add opentelemetry.io docs

### PR DESCRIPTION
**Pull request by austinlparker**
Per https://github.com/open-telemetry/opentelemetry.io/issues/472, we're adding the documentation content from the website to each repository. After a release of that requires documentation changes, please make an issue or PR in the website repository in order to have the changes mirrored.

### CODE REVIEW

File: website_docs/automatic_instrumentation.md [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+
+#### Jaeger exporter
+
+> Currently only supports gRPC.
+
+| System property                   | Environment variable              | Purpose                                                                                            |
+|-----------------------------------|-----------------------------------|----------------------------------------------------------------------------------------------------|
+| otel.exporter=jaeger              | OTEL_EXPORTER=jaeger              | To select Jaeger exporter                                                                          |
+| otel.exporter.jaeger.endpoint     | OTEL_EXPORTER_JAEGER_ENDPOINT     | The Jaeger endpoint to connect to, default is "localhost:14250", currently only gRPC is supported. |
+| otel.exporter.jaeger.service.name | OTEL_EXPORTER_JAEGER_SERVICE_NAME | The service name of this JVM instance, default is "unknown".                                       |
```
Comment (jkwatson) [inline website_docs/automatic_instrumentation.md:172]: this line should be deleted, as it is no longer a valid property.

Comment (jkwatson) [inline website_docs/automatic_instrumentation.md:172]:
```suggestion
```

File: website_docs/automatic_instrumentation.md [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+
+#### Zipkin exporter
+
+| System property                   | Environment variable              | Purpose                                                                                                               |
+|-----------------------------------|-----------------------------------|-----------------------------------------------------------------------------------------------------------------------|
+| otel.exporter=zipkin              | OTEL_EXPORTER=zipkin              | To select Zipkin exporter                                                                                             |
+| otel.exporter.zipkin.endpoint     | OTEL_EXPORTER_ZIPKIN_ENDPOINT     | The Zipkin endpoint to connect to, default is "http://localhost:9411/api/v2/spans". Currently only HTTP is supported. |
+| otel.exporter.zipkin.service.name | OTEL_EXPORTER_ZIPKIN_SERVICE_NAME | The service name of this JVM instance, default is "unknown".                                                          |
```
Comment (jkwatson) [inline website_docs/automatic_instrumentation.md:180]:
```suggestion
```

File: website_docs/automatic_instrumentation.md [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+
+#### Prometheus exporter
+
+| System property               | Environment variable          | Purpose                                                                            |
+|-------------------------------|-------------------------------|------------------------------------------------------------------------------------|
+| otel.exporter=prometheus      | OTEL_EXPORTER=prometheus      | To select Prometheus exporter                                                      |
```
Comment (jkwatson) [inline website_docs/automatic_instrumentation.md:186]: this is no longer a valid property. should be `otel.metrics.exporter` I think

File: website_docs/automatic_instrumentation.md [hunk 4]
```diff
@@ -0,0 +1,389 @@
...
+| otel.exporter.prometheus.host | OTEL_EXPORTER_PROMETHEUS_HOST | The local address used to bind the prometheus metric server, defaults to "0.0.0.0" |
+
+#### Logging exporter
+
+> The logging exporter simply prints the name of the span along with its
+attributes to stdout. It is used mainly for testing and debugging.
+
+| System property              | Environment variable         | Purpose                                                                      |
+|------------------------------|------------------------------|------------------------------------------------------------------------------|
+| otel.exporter=logging        | OTEL_EXPORTER=logging        | To select logging exporter                                                   |
```
Comment (jkwatson) [inline website_docs/automatic_instrumentation.md:197]: this should be `otel.traces/metrics.exporter`

File: website_docs/automatic_instrumentation.md [hunk 5]
```diff
@@ -0,0 +1,389 @@
...
+### Trace config
+
+| System property                 | Environment variable            | Purpose                                              |
+|---------------------------------|---------------------------------|------------------------------------------------------|
+| otel.config.sampler.probability | OTEL_CONFIG_SAMPLER_PROBABILITY | Sampling probability between 0 and 1 (default: 1)    |
+| otel.config.max.attrs           | OTEL_CONFIG_MAX_ATTRS           | Maximum number of attributes per span (default: 32)  |
+| otel.config.max.events          | OTEL_CONFIG_MAX_EVENTS          | Maximum number of events per span (default: 128)     |
+| otel.config.max.links           | OTEL_CONFIG_MAX_LINKS           | Maximum number of links per span (default: 32)       |
+| otel.config.max.event.attrs     | OTEL_CONFIG_MAX_EVENT_ATTRS     | Maximum number of attributes per event (default: 32) |
+| otel.config.max.link.attrs      | OTEL_CONFIG_MAX_LINK_ATTRS      | Maximum number of attributes per link (default: 32)  |
```
Comment (jkwatson) [inline website_docs/automatic_instrumentation.md:257]: these 6 properties are no longer valid. The ones in the autoconfigure README should be used.

File: website_docs/automatic_instrumentation.md [hunk 6]
```diff
@@ -0,0 +1,389 @@
...
+
+> This is highly advanced behavior and still in the prototyping phase. It may
+change drastically or be removed completely. Use with caution
+
+The OpenTelemetry API exposes SPI
+[hooks](https://github.com/open-telemetry/opentelemetry-java/blob/main/api/src/main/java/io/opentelemetry/trace/spi/TracerProviderFactory.java)
+for customizing its behavior, such as the `Resource` attached to spans or the
+`Sampler`.
+
+Because the auto instrumentation runs in a separate classpath than the
```
Comment (jkwatson) [inline website_docs/automatic_instrumentation.md:275]: none of the auto-configuration docs belong in this repo.

File: website_docs/automatic_instrumentation.md [hunk 7]
```diff
@@ -0,0 +1,389 @@
...
+### Dependency
+
+#### Maven
+
+```xml
+  <dependencies>
+    <dependency>
+      <groupId>io.opentelemetry</groupId>
+      <artifactId>opentelemetry-api</artifactId>
+      <version>0.10.0</version>
```
Comment (jkwatson) [inline website_docs/automatic_instrumentation.md:314]: this is very out of date. we should be recommending use of the pom, and, of course, version 1.0.1

File: website_docs/automatic_instrumentation.md [hunk 8]
```diff
@@ -0,0 +1,389 @@
...
+
+#### Gradle
+
+```groovy
+dependencies {
+    compile('io.opentelemetry:opentelemetry-api:0.10.0')
+}
+```
+
+### Configure `@WithSpan` annotation
```
Comment (jkwatson) [inline website_docs/automatic_instrumentation.md:327]: this is for the instrumentation project

File: website_docs/automatic_instrumentation.md [hunk 9]
```diff
@@ -0,0 +1,389 @@
...
+
+##### Groovy
+
+```groovy
+dependencies {
+    compile('io.opentelemetry:opentelemetry-extension-auto-annotations:0.10.0')
+}
+```
+
+#### Suppressing `@WithSpan` instrumentation
```
Comment (jkwatson) [inline website_docs/automatic_instrumentation.md:368]: this is more instrumentation agent docs that don't belong in this repo.

**Comment by jkwatson** [comment]
This content has a mishmash of API,SDK and auto-instrumentation agent documentation in it. I think we shouldn't have auto-instrumentation docs in this repo, but I don't know how to generate a unified set of docs for the website if they are split. @austinlparker Any thoughts on re-structuring the website docs so they can be maintained independently, between API/SDK and the agent/instrumentation repositories?

File: website_docs/_index.md [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+- [opentelemetry-java-contrib](https://github.com/open-telemetry/opentelemetry-java-contrib):
+  Provides helpful libraries and standalone OpenTelemetry-based utilities that
+  don't fit the express scope of the OpenTelemetry Java or Java Instrumentation
+  projects. For example, JMX metric gathering.
+
+## opentelemetry-java
+
+| Traces | Metrics | Logs         |
+| ------ | ------- | ------------ |
+| Beta   | Alpha   | Experimental |
```
Comment (jkwatson) [inline website_docs/_index.md:27]:
```suggestion
| GA   | Alpha   | Experimental |
```

**Comment by jkwatson** [comment]
> There's nothing that says 'Java' and 'Java Automatic Instrumentation' can't be different headings, but I think from an end-user's perspective looking for "getting started" guides we could have a website_docs folder in the auto-instrumentation repo and have that be committed to the same place in the website repo?

So, maybe it would be easiest for me to take the content of this PR and strip out the non-relevant-to-this-repo contents (and fix up the out-of-dateness), and submit it as a fresh PR for your review?

**Comment by austinlparker** [comment]
I don't really need to review your changes to it... I'm just trying to copy the current state of the website docs into each SIG repo, then you can make whatever changes you desire then we can apply those changes back to the website.

**Comment by austinlparker** [comment]
My next step is creating a GitHub action that will automatically make that PR for you when a new release is made.

**Comment by jkwatson** [comment]
> I don't really need to review your changes to it... I'm just trying to copy the current state of the website docs into each SIG repo, then you can make whatever changes you desire then we can apply those changes back to the website.

ah, ok. I guess the thought it that it will be ok to have it be out of date/wrong initially, but it could be cleaned up and then pushed?  Given that, can you put a disclaimer at the top of the file that it is not current, and then I'll remove that on my PR to clean it up?

**Comment by austinlparker** [comment]
Done

**Review by jkwatson — APPROVED** [review]
Thanks! once merged, I'll work on getting it current, and removing non-otel-java stuff.

**Review by anuraaga — APPROVED** [review]

**Comment by jkwatson** [comment]
merging. I'll pick up the cleanup later today.

**Pull request merged**